Parse/src/main/java/com/parse/ParseObject.java

```diff
@@ -516,41 +491,11 @@ private static boolean isAccessible(Member m) {
    *          The subclass type to register.
    */
   public static void registerSubclass(Class<? extends ParseObject> subclass) {
-    String className = getClassName(subclass);
-    if (className == null) {
-      throw new IllegalArgumentException("No ParseClassName annotation provided on " + subclass);
-    }
-    if (subclass.getDeclaredConstructors().length > 0) {
-      try {
-        if (!isAccessible(subclass.getDeclaredConstructor())) {
-          throw new IllegalArgumentException("Default constructor for " + subclass
-              + " is not accessible.");
-        }
-      } catch (NoSuchMethodException e) {
-        throw new IllegalArgumentException("No default constructor provided for " + subclass);
-      }
-    }
-    Class<? extends ParseObject> oldValue = objectTypes.get(className);
-    if (oldValue != null && subclass.isAssignableFrom(oldValue)) {
-      // The old class was already more descendant than the new subclass type. No-op.
-      return;
-    }
-    objectTypes.put(className, subclass);
-    if (oldValue != null && !subclass.equals(oldValue)) {
-      if (className.equals(getClassName(ParseUser.class))) {
-        ParseUser.getCurrentUserController().clearFromMemory();
```

Comment:
Similar to the state we have on .NET, we lose this logic during this decouple. Thoughts on a good cross-platform and extensible way to have a hook when a certain class is unregistered?